README.md

```diff
@@ -145,104 +147,129 @@ Instead of *just* looking for `.goldengrahamsrc` (no extension), it will also lo
+By default, it's asynchronous.
 
 ##### transform
 
-Type: `Function`
+Type: `(Result) => Promise<Result> | Result`.
 
-A function that transforms the parsed configuration. Receives the result object with `config` and `filepath` properties.
+A function that transforms the parsed configuration. Receives the [result] object.
 
-If the option `sync` is `false` (default), the function must return a Promise that resolves with the transformed result.
-If the option `sync` is `true`, though, `transform` should be a synchronous function which returns the transformed result.
+If the option [`sync`] is `false` (default), the function must return a Promise that resolves with the transformed result.
+If the option [`sync`] is `true`, the function must be synchronous and return the transformed result.
 
 The reason you might use this option instead of simply applying your transform function some other way is that *the transformed result will be cached*. If your transformation involves additional filesystem I/O or other potentially slow processing, you can use this option to avoid repeating those steps every time a given configuration is searched or loaded.
 
 ##### configPath
 
-Type: `string`
+Type: `string`.
 
-If provided, cosmiconfig's [`load()`](#loadconfigpath) will default to loading a config from this path.  Passing a string parameter will override this option.
+If provided, the explorer's [`load()`] will default to loading a config from this path. Passing a string parameter to [`load()`] will override this option.
 
 ##### format
 
 Type: `'json' | 'yaml' | 'js'`
 
-The expected file format for the config when running [`load()`](#loadconfigpath).
+The expected file format for the config when running [`load()`].
 
 If not specified, cosmiconfig will try to infer the format using the extension name (if it has one).
 In the event that the file does not have an extension or the extension is unrecognized, cosmiconfig will try to parse it as a JSON, YAML, or JS file.
 
-### Instance methods (on `explorer`)
+### explorer.search
 
-#### `search([searchPath][, searchOptions])`
+```js
+explorer.search([searchPath][, searchOptions])
+```
 
-Searches for a configuration file. Returns a Promise that resolves with `null`, if nothing is found, or an object with two properties:
-- `config`: The loaded and parsed configuration.
-- `filepath`: The filepath where this configuration was found.
+Searches for a configuration file. In async mode (default), returns a Promise that resolves a [result] object or with `null`, if no configuration file is found. In [`sync`] mode, returns a [result] object or `null`.
 
-When the [search option](#searchoptions) `ignoreEmpty` is set to false and an empty configuration file is found, the result will have `config` set to `undefined` and an additional property `isEmpty` set to true.
+So let's say `const explorer = cosmiconfig('goldengrahams');` — here's how `explorer.search` will work:
```

Comment:
Change to "here's how `explorer.search` works" to be consistent with the rest of the explanation.